Feature/40 question summary page

nmenezes0 opened the pull request:
## Context

<!-- Why are you making this change? What might surprise someone about it? -->

Add a very basic question summary page (content and styling to be improved!).

## Changes proposed in this pull request

<!-- If there are UI changes, please include Before and After screenshots. -->

Add a very basic question summary page - text of the question and list of themes.

This involves:
* Adding a Theme factory
* Adding a new page for the question summary

Additionally, changes in the migrations file are just from running ruff (code checks added in a separate PR).

![image](https://github.com/i-dot-ai/consultation-analyser/assets/77671865/4ab89ae3-5f99-41b8-ad84-bbdce6c62c8b)

TODO in a separate PR - styling.

## Guidance to review

<!-- How could someone else check this work? Which parts do you want more feedback on? -->

* Generate some data by running the management command: `poetry run python manage.py generate_dummy_data`.
* Go to the URL: `consultation/<str:consultation_slug>/section/<str:section_slug>/question-summary/<str:question_slug>/` - and check question summary and some themes display (there may be duplicates, haven't gone as far as getting counts).
(Get slugs from the database - pick a question and get the slug for that, the related section, and the consultation related to that section.)
* General code review.


## Link to JIRA ticket

<!-- e.g. https://technologyprogramme.atlassian.net/jira/software/c/projects/ER/boards/346?selectedIssue=ER-87 -->
https://technologyprogramme.atlassian.net/browse/CON-40


## Things to check

- [X] I have added any new ENV vars in all deployed environments and updated the `.env.example` and `.env.test` files in the repo - N/A

duncanjbrown commented on consultation_analyser/consultations/views.py:
```diff
@@ -1,12 +1,22 @@
     return render(request, "show_question.html", context)
+
+
+def question_summary(request: HttpRequest, consultation_slug: str, section_slug: str, question_slug: str):
```
I don't think this is a different view to the one above?

nmenezes0 replied:
Now renamed to `show_question` and the other view removed.

duncanjbrown commented on tests/integration/test_question_pages.py:
```diff
@@ -0,0 +1,39 @@
+        section=section,
+        slug="question-slug",
+        text="Is this an interesting question?",
+        multiple_choice_options=[],
```
I think this is blank by default?

nmenezes0 replied:
By default it gets whatever was in the YAML file, but we're doing this differently now anyway.

nmenezes0 commented on consultation_analyser/consultations/urls.py:
```diff
@@ -1,4 +1,12 @@
+    path("", views.home),
+    path("questions/<str:slug>", views.show_question),
+    path(
+        "consultation/<str:consultation_slug>/section/<str:section_slug>/question-summary/<str:question_slug>/",
```
Yes, now changes to `question` (and the other URL removed).

nmenezes0 commented on tests/integration/test_question_pages.py:
```diff
@@ -0,0 +1,39 @@
+    assert "How should funding be allocated?" in str(response.content)
+
+
+def generate_question_with_themes():
```
Done!

nmenezes0 commented:
Ready for re-review. Will rebase on top of main once the jinja branch is in.

nmenezes0 commented:
Changes now made - updated URL and name of view, deleted old question page (was there as a test), added a factory to create Answers, Themes etc for test.

duncanjbrown reviewed (commented):
Looking good! few comments inline

nmenezes0 commented on consultation_analyser/consultations/migrations/0001_initial.py:
```diff
@@ -6,105 +6,124 @@
     operations = [
         migrations.CreateModel(
-            name='Consultation',
+            name="Consultation",
```
I think the one with the double quotation marks is the correct one - I wasn't running pre-commit before (there's a separate PR to improve the repo set-up including installing pre-commit).

nmenezes0 replied:
This change now gone on rebase.

nmenezes0 commented on .github/workflows/test.yml:
```diff
@@ -41,7 +41,16 @@ jobs:
+        with:
+          node-version: '14'
+
+      - name: Install dependencies
```
Updated

nmenezes0 commented on .github/workflows/test.yml:
```diff
@@ -41,7 +41,16 @@ jobs:
       - name: pytest
-        run: poetry run pytest
+        run:
+          poetry run pytest
```
Updated

nmenezes0 commented on consultation_analyser/consultations/urls.py:
```diff
@@ -1,4 +1,11 @@
+urlpatterns = [
+    path("", views.home),
+    path(
+        "consultation/<str:consultation_slug>/section/<str:section_slug>/question/<str:question_slug>/",
```
Updated to `consultations/<str:consultation_slug>/sections/<str:section_slug>/questions/<str:question_slug>` (and trailing `/` removed).

nmenezes0 commented on tests/factories.py:
```diff
@@ -52,20 +48,48 @@ class Meta:
     slug = faker.slug()
 
+    class Params:
+        fixed_slug = factory.Trait(
```
Now much simplified (just take the generated Answer, Question etc. and use that in the test).

nmenezes0 commented on consultation_analyser/consultations/jinja2/show_question.html:
```diff
@@ -1 +1,14 @@
+
+{% block content %}
+
+  <h1 class="govuk-heading-xl">Question summary - dummy page</h1>
```
Have copied some of the prototype (we don't have data for all the elements yet):
![image](https://github.com/i-dot-ai/consultation-analyser/assets/77671865/10697836-d91b-43fc-ab0e-7b1f9c4710a0)

(Kev also said he'll work on improving this once it's merged!)

duncanjbrown reviewed (commented):
Looks vg! pls could you rebase and tidy commits?
